eth_utils/abi.py

```diff
@@ -1,68 +1,793 @@
+from collections import (
+    abc,
+)
+import copy
+import itertools
+import re
 from typing import (
     Any,
     Dict,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    cast,
+)
+
+from eth_typing import (
+    ABI,
+    ABICallable,
+    ABIComponent,
+    ABIElement,
+    ABIError,
+    ABIEvent,
+    ABIFunction,
+)
+
+from eth_utils.types import (
+    is_list_like,
 )
 
 from .crypto import (
     keccak,
 )
 
 
-def collapse_if_tuple(abi: Dict[str, Any]) -> str:
+def _abi_input_types(
+    abi_inputs: Optional[Sequence[Union[ABIComponent, str]]] = None,
```

Comment:
Is there a utility in keeping this as a kwarg and allowing it to be `None`? It doesn't seem useful at a glance. Perhaps you have a bit more context here.

Reply:
This implementation does seem unclear since it is running through the `join` needlessly in the case when `None` is passed. It should just return an empty string.

As I understand, Python convention usually sets the arg to `None` and sets a default value within the function itself. I think that's what I was doing here, but may not be the best approach.

Do you think it would be better to only call this function when there are inputs? That logic could exist in `abi_to_signature`. Let me know your thoughts!

Reply:
Yeah, accounting for getting ABI input types from nothing seems like an unnecessary bit of caution. I'd expect this to be an arg like the other methods, rather than a kwarg. Unless there's some utility I'm not seeing.